---
title: "Leukemia microarray data from Yeoh et. al. (2002)"
author: "Group 20: Mikkel Spallou Eriksen (s122455), Rut Mas de Les Valls (s203003), Anna Oliver Almirall (s202436), Paul Jeremy Simon (s202592), Elisa Catafal Tardos (s210415)"
date: "11/05/2021"
output: 
  ioslides_presentation:
    logo: dtu.png
    df_print: tibble
    widescreen: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(eval = FALSE)
knitr::opts_knit$set(root.dir = '../')
```

## Introduction

- The paper from which the dataset is taken is "Classification, subtype discovery, and prediction of outcome in pediatric acute lymphoblastic leukemia by gene expression profiling" written by Yeoh et. al. in 2002 in Cell. 
- The study is about pediatric acute lymphoblastic leukemia (ALL), a blood cancer with different subtypes.
- Each subtype has specific characteristics including its response to treatment.
- The goal of the project is to **assess whether there is a link between gene expression and the type of ALL**.


## Materials

- Raw files:
  - raw_stjude_x.tsv.gz, a matrix of 12,625 rows (gene probes) and 327 columns (indivuals)
  - raw_stjude_xrows.tsv.gz with the names of the probes
  - raw_bcr.xls, a Excel spreadsheet from which we are interested in the first 2 columns giving the equivalence between the probe name and the gene name.


- R with Tidyverse !

## Methods
- A regular expression to extract the gene name from the Excel column 
```{r regexp}
gene_pattern <- "\\w{1,2}\\d{5,6}"
```

## Methods
- The stringr package to tidy the individuals' names
```{r id_extraction}
x_clean <- x_clean %>% 
  separate(col = "id", 
           into = c("leukemia", NA, NA, NA), 
           sep = "-")
```
From
```{r individual_name}
"BCR-ABL-#7"
```
we extract
```{r individual_extract}
"BCR"
```
which is a type of ALL.

## Methods
- An apply_t_test function to perform selection of the best genes.
```{r apply_ttest}
apply_t_test = function(data){
  cancer_types <- c("BCR", "E2A", "Hyperdip", "MLL", "T", "TEL")
  my_list <- map(cancer_types,
                 ~ t.test(data %>%
                            filter(leukemia == .x) %>% 
                            pull(expr_level),
                          data %>%
                            filter(leukemia != .x) %>% 
                            pull(expr_level)) %>% 
                   tidy() %>% 
                   pull(statistic))
  names(my_list) <- cancer_types
  return(my_list)
}
```


## Methods
- The ggplot2 package to create different plot for the dataset
```{r plots}
# Example of plot
```

## Results
### Kmeans with top 40 genes
```{r kmeansimport, echo = FALSE, eval = TRUE, results = "hide", message = FALSE}
source("R/06_kmeans.R")
```

```{r kmeanstable, eval=TRUE, echo=FALSE, message=FALSE, paged.print=TRUE}
clusters_top40
```

## Results
### Kmeans with top 40 genes



## Results
### PCA

## Results
```{r, echo=FALSE, out.width="70%", error=TRUE, eval=TRUE, fig.cap = "Heat map for the 10 most upregulated genes"}
knitr::include_graphics(path = str_c(getwd(),"/results/05_heatmap_1_pos.png"))
```

## Results
### Shiny app
[Link](https://paul-simon.shinyapps.io/yeoh_plots/)

## Discussion
- By genotyping a patient, it can be possible to to know which type of cancer he/she has and adapt the treatment
- In the paper, the show that it works ... (***DETAILS***)
