---
title: "Leukemia microarray data from Yeoh et. al. (2002)"
author: "Group 20: Mikkel Spallou Eriksen (s122455), Rut Mas de Les Valls (s203003), Anna Oliver Almirall (s202436), Paul Jeremy Simon (s202592), Elisa Catafal Tardos (s210415)"
date: "11/05/2021"
output: 
  ioslides_presentation:
    logo: dtu.png
    df_print: tibble
    widescreen: true
    smaller: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(tidyverse)
```

## Introduction

- The paper from which the dataset is taken is "Classification, subtype discovery, and prediction of outcome in pediatric acute lymphoblastic leukemia by gene expression profiling" written by Yeoh et. al. in 2002 in Cell. 
- The study is about pediatric acute lymphoblastic leukemia (ALL), a blood cancer with different subtypes.
- Each subtype has specific characteristics including its response to treatment.
- The goal of the project is to **assess whether there is a link between gene expression and the type of ALL**.


## Introduction
- Leukemia types


## Materials

- Raw files:
  - raw_stjude_x.tsv.gz, a matrix of 12,625 rows (gene probes) and 327 columns (indivuals);
  - raw_stjude_xrows.tsv.gz with the names of the probes;
  - raw_bcr.xls, an Excel spreadsheet from which we are interested in the first 2 columns giving the equivalence between the probe name and the gene name.


- R with Tidyverse !


## Methods | Outline of the project
```{r outline, echo=FALSE, out.width="85%", fig.align = 'center'}
knitr::include_graphics(path = "diagram.png")
```

## Methods | From raw data ...
- raw_stjude_x.tsv
```{r echo=TRUE, eval=FALSE}
# A tibble: 12,625 x 327
   `BCR-ABL-#1` `BCR-ABL-#2` `BCR-ABL-#3` `BCR-ABL-#4` `BCR-ABL-#5` `BCR-ABL-#6` `BCR-ABL-#7` `BCR-ABL-#8`
          <dbl>        <dbl>        <dbl>        <dbl>        <dbl>        <dbl>        <dbl>        <dbl>
 1        1.45         1.34         1.49         1.47         1.49         1.45         1.46         1.51 
 2        0.984        1.09         1.10         0.975        0.933        0.968        0.887        0.983
 3        0.892        0.897        0.807        0.969        0.797        0.979        0.919        0.902
 4        1.23         1.28         1.16         1.29         1.18         1.20         1.21         1.24 
 5        1.20         1.12         1.12         1.17         1.11         1.10         1.11         1.06 
```


## Methods | From raw data ...
- raw_stjude_xrows.tsv
```{r echo=TRUE, eval=FALSE}
# A tibble: 12,625 x 1
   value    
   <chr>    
 1 1000_at  
 2 1001_at  
 3 1002_f_at
 4 1003_s_at
 5 1004_at  
```


## Methods | From raw data ...
- raw_bcr.xls
```{r echo=TRUE, eval=FALSE}
# A tibble: 12,626 x 32
   `Probe set` Descriptions `BCR-ABL-C1...3` `BCR-ABL-C1...4` `BCR-ABL-R1...5` `BCR-ABL-R1...6` `BCR-ABL-R2...7`
   <chr>       <chr>        <chr>            <chr>            <chr>            <chr>            <chr>           
 1 NA          NA           Avg Diff         Abs Call         Avg Diff         Abs Call         Avg Diff        
 2 AFFX-MurIL~ M16762 Mous~ -2597.800000000~ A                175.69999999999~ A                1249.2          
 3 AFFX-MurIL~ M37897 Mous~ -404.1000000000~ A                -512.5           A                -57.60000000000~
 4 AFFX-MurIL~ M25892 Mus ~ 2143.3000000000~ A                737.70000000000~ A                -941.1000000000~
 5 AFFX-MurFA~ M83649 Mus ~ -1697.7          A                1082.3           P                -276.5          
```


## Methods | ... to tidy data
- 02_stjude_clean.tsv.gz
```{r echo=TRUE, eval=FALSE}
# A tibble: 248 x 12,627
   sampleID leukemia Y08200_2 X60188 X60957 X65962 X68149_1 X68149_2 X68277 X07820 U48705 U50648 U51004 Y09305
   <chr>    <chr>       <dbl>  <dbl>  <dbl>  <dbl>    <dbl>    <dbl>  <dbl>  <dbl>  <dbl>  <dbl>  <dbl>  <dbl>
 1 s_1      BCR          1.51   1.45  0.984  0.892     1.23     1.20   1.67  0.870   1.25   2.48   2.39  1.20 
 2 s_2      BCR          1.54   1.34  1.09   0.897     1.28     1.12   1.87  0.910   1.67   2.35   2.09  1.16 
 3 s_3      BCR          1.60   1.49  1.10   0.807     1.16     1.12   2.63  0.866   1.51   2.20   2.47  1.23 
 4 s_4      BCR          1.64   1.47  0.975  0.969     1.29     1.17   2.50  0.910   1.69   2.72   2.26  0.959
 5 s_5      BCR          1.70   1.49  0.933  0.797     1.18     1.11   1.65  0.869   1.63   2.78   1.88  1.08 
```



## Methods
- A regular expression to extract the gene name from the Excel column 
```{r regexp, echo=TRUE, eval=FALSE}
gene_pattern <- "\\w{1,2}\\d{5,6}"
```


## Methods
- The stringr package to tidy the individuals' names
```{r id_extraction, echo=TRUE, eval=FALSE}
x_clean <- x_clean %>% 
  separate(col = "id", 
           into = c("leukemia"), 
           sep = "-")
```
From
```{r individual_name, echo=TRUE, eval=FALSE}
"BCR-ABL-#7"
```
we extract
```{r individual_extract, echo=TRUE, eval=FALSE}
"BCR"
```
which is a type of ALL.


## Methods
- An apply_t_test function to perform selection of the best genes.
```{r apply_ttest, echo=TRUE, eval=FALSE}
apply_t_test = function(data){
  cancer_types <- c("BCR", "E2A", "Hyperdip", "MLL", "T", "TEL")
  my_list <- map(cancer_types,
                 ~ t.test(data %>%
                            filter(leukemia == .x) %>% 
                            pull(expr_level),
                          data %>%
                            filter(leukemia != .x) %>% 
                            pull(expr_level)) %>% 
                   tidy() %>% 
                   pull(statistic))
  names(my_list) <- cancer_types
  return(my_list)
}
```
- Paul will add something here


## Results | Kmeans (all genes)
```{r kmeanstableall, message=FALSE}
kmeans_clusters_all <- read_tsv(file = "../results/06_kmeans_table_all.tsv")
kmeans_clusters_all %>% as_tibble
```

## Results | Kmeans (all genes)
```{r kmeansplotall, out.width="60%", fig.align ='center'}
knitr::include_graphics(path = "../results/06_kmeans_barplot_all.png")
```


## Results | Kmeans (top 40 genes)
```{r kmeanstable40, eval=TRUE, message = FALSE}
kmeans_clusters_top40 <- read_tsv(file = "../results/06_kmeans_table_top40.tsv")
kmeans_clusters_top40 %>% as_tibble
```

## Results | Kmeans (top 40 genes)
```{r kmeansplot40, out.width="60%", fig.align ='center'}
knitr::include_graphics(path = "../results/06_kmeans_barplot_top40.png")
```


## Results | PCA
```{r pcaprojectionall, out.width="60%", fig.align ='center'}
knitr::include_graphics(path = "../results/07_pca_all.png")
```


## Results | PCA
```{r pcavariancenall, out.width="60%", fig.align ='center'}
knitr::include_graphics(path = "../results/07_pca_var_all.png")
```


## Results | PCA
```{r pcaprojectiontop, out.width="60%", fig.align ='center'}
knitr::include_graphics(path = "../results/07_pca_top40.png")
```


## Results | PCA
```{r pcavariancetop, out.width="60%", fig.align ='center'}
knitr::include_graphics(path = "../results/07_pca_var_top40.png")
```

## Results | Reproducing _Figure 3_ from the paper
```{r heatmappaper, out.width="75%", fig.align ='center'}
knitr::include_graphics(path = "paper_heatmap.png")
```

## Results | Reproducing _Figure 3_ from the paper:  __Top positively correlated genes__

```{r heatmap1, out.width="100%", fig.align ='center'}
knitr::include_graphics(path = "../results/05_heatmap_1_pos.png")
```


## Results| Reproducing _Figure 3_ from the paper: __Top 10 correlated genes__
```{r heatmap2, out.width="100%", fig.align ='center'}
knitr::include_graphics(path = "../results/05_heatmap_2_top10.png")
```


## Results | Our suggestion:  __Top positively correlated genes__
```{r heatmap3, out.width="100%", fig.align ='center'}
knitr::include_graphics(path = "../results/05_heatmap_3_global_pos.png")
```


## Results | Our suggestion: __Top 10 correlated genes__
```{r heatmap4, out.width="100%", fig.align ='center'}
knitr::include_graphics(path = "../results/05_heatmap_4_global_top10.png")
```


---
```{r shinyapp}
knitr::include_app(url = "https://paul-simon.shinyapps.io/yeoh_plots/")
```

[Alternatively, the link](https://paul-simon.shinyapps.io/yeoh_plots/)


## Discussion
- Tidying the data from raw was successful.
- T-test was used to select the top 40 most differentialy expressed genes.
- The PCA and Kmeans separated the types slightly better when the top 40 genes of each type were selected.
- This demonstrates that the genes selected were representative of their type.
- By genotyping a patient, it can be possible to to know which type of cancer he/she has and adapt the treatment.
- In the paper, the results of the microarray are compared to RNA levels from RT-PCR and they show that there is high correlation between the two.
